app/src/main/java/com/google/android/gms/samples/wallet/CheckoutActivity.java

```diff
@@ -69,37 +73,17 @@
     private ProgressDialog mProgressDialog;
 
     @Override
-    protected void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_checkout);
-
-        // [START basic_google_api_client]
-        mGoogleApiClient = new GoogleApiClient.Builder(this)
-                .addApi(Wallet.API, new Wallet.WalletOptions.Builder()
-                        .setEnvironment(Constants.WALLET_ENVIRONMENT)
-                        .build())
-                .enableAutoManage(this, this)
+    public void onConnected(@Nullable Bundle bundle) {
+        // Configure isReadyToPay to return true for default networks
+        IsReadyToPayRequest isReadyToPayRequest = IsReadyToPayRequest.newBuilder()
                 .build();
-        // [END basic_google_api_client]
-
-        mItemId = getIntent().getIntExtra(Constants.EXTRA_ITEM_ID, 0);
-        mReturnToShopping = (Button) findViewById(R.id.button_return_to_shopping);
-        mReturnToShopping.setOnClickListener(this);
-        mContinueCheckout = (Button) findViewById(R.id.button_regular_checkout);
-        mContinueCheckout.setOnClickListener(this);
-
-        mStripeCheckbox = (CheckBox) findViewById(R.id.checkbox_stripe);
-        mStripeCheckbox.setOnCheckedChangeListener(this);
-
         // Check if user is ready to use Android Pay
         // [START is_ready_to_pay]
-        showProgressDialog();
-        Wallet.Payments.isReadyToPay(mGoogleApiClient).setResultCallback(
+        Wallet.Payments.isReadyToPay(mGoogleApiClient, isReadyToPayRequest).setResultCallback(
```

Comment:
Just FYI when you use GoogleApiClient any "read-only" requests (like isReadyToPay) are queued even if you call them before onConnected, the ResultCallback will only fire after connection.

Is there some other reason you want to move this into onConnected that I am missing?